libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -912,15 +1016,22 @@ bool HTTPClient::connect(void)
 
     DEBUG_HTTPCLIENT("[HTTP-Client] connected to %s:%u\n", _host.c_str(), _port);
 
-    if (!_transportTraits->verify(*_tcp, _host.c_str())) {
+#ifdef KEEP_PRESENT_API
+    if (_tcpDeprecated && _transportTraits && !_transportTraits->verify(*_tcpDeprecated, _host.c_str())) {
         DEBUG_HTTPCLIENT("[HTTP-Client] transport level verify failed\n");
         _tcp->stop();
         return false;
     }
+#endif
 
 
 #ifdef ESP8266
-    _tcp->setNoDelay(true);
+#ifdef KEEP_PRESENT_API
+    if(_tcpDeprecated)
+        _tcpDeprecated->setNoDelay(true);
```

Comment:
Wouldn't you want to set it on `_tcp` as well (outside this `#define`, of course)?

Reply:
Client::setNoDelay() does not exist. If I change Client in WiFiClient its is possible, see my remark later

Reply:
Now setNoDelay() is alway called